lib/matplotlib/tests/test_axes.py

```diff
@@ -1055,6 +1056,27 @@ def test_hexbin_mincnt_behavior_upon_C_parameter(fig_test, fig_ref):
     ax_test.set_facecolor("green")
 
 
+@image_comparison(['hexbin_symlog.png'], style='mpl20')
+def test_hexbin_symlog():
+    np.random.seed(19680801)
+    base = 4
+    linthresh = base**0
+    linscale = 1
+    symlog_transform = mscale.SymmetricalLogTransform(base,
+                                                      linthresh,
+                                                      linscale).inverted()
+    n = 100000
+    x = np.random.standard_normal(n)
+    y = x * (2*np.random.randint(2, size=n) - 1)
+    y = symlog_transform.transform_non_affine(y)
+
+    fig, ax = plt.subplots()
+    h = ax.hexbin(x, y, yscale=('symlog', {'base': base,
+                  'linthresh': linthresh, 'linscale': linscale}), bins='log',
+                  marginals=True, reduce_C_function=np.sum)
```

Comment:
Can we have this test out tuples/strings for `xscale` as well? Perhaps use a 2x2 Axes and then plot xscale/yscale with plain 'symlog' and with a tuple. That should fix code coverage, and also confirm all cases are working.